lib/galaxy/datatypes/data.py

```diff
@@ -994,6 +994,45 @@ def get_visualizations(self, dataset):
         return ['phyloviz']
 
 
+class IQTree(Text):
+    """IQ-TREE format"""
+    file_ext = 'iqtree'
+
+    def __init__(self, **kwd):
+        """Init mixed format"""
+        Text.__init__(self, **kwd)
+
+    def init_meta(self, dataset, copy_from=None):
+        Text.init_meta(self, dataset, copy_from=copy_from)
+
+    def sniff(self, filename):
+        """
+        Detect the IQTree file
+
+        Scattered text file containing various headers and data
+        types.
+
+        >>> from galaxy.datatypes.sniff import get_test_fname
+        >>> fname = get_test_fname('example.iqtree')
+        >>> IQTree().sniff(fname)
+        True
+
+
+        >>> fname = get_test_fname('temp.txt')
+        >>> IQTree().sniff(fname)
+        False
+
+
+        >>> fname = get_test_fname('test_tab1.tabular')
+        >>> IQTree().sniff(fname)
+        False
+        """
+        with open(filename, 'r') as fio:
+            return fio.readline().split()[0] == "IQ-TREE"
```

Comment:
Can you use: `fio.read(7)`. This will prevent reading files with a huge lines entirely into memory.